cmd/release-controller/main.go

```diff
@@ -432,6 +432,13 @@ func (o *options) Run() error {
 					if !ok {
 						continue
 					}
+					jobArchitecture, ok := job.GetAnnotations()[releaseAnnotationArchitecture]
```

Comment:
split out job.GetAnnotations() at the beginning of this method and have each check read from it.  Fetching the method 3 times is ugly.